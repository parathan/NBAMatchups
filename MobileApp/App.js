import { NavigationContainer } from '@react-navigation/native';
import { createNativeStackNavigator } from '@react-navigation/native-stack';

import HomeScreen from './src/screens/HomeScreen';
import MatchupsScreen from './src/screens/MatchupsScreen';
import DashboardScreen from './src/screens/DashboardScreen';
import PredictiveScreen from './src/screens/PredictiveScreen';

const Stack = createNativeStackNavigator();

export default function App() {
  return (
    <NavigationContainer>
      <Stack.Navigator>
        <Stack.Screen name="Home" component={HomeScreen} />
        <Stack.Screen name="Matchups" component={MatchupsScreen} />
        <Stack.Screen name="Dashboard" component={DashboardScreen} />
        <Stack.Screen name="Prediction" component={PredictiveScreen} />
      </Stack.Navigator>
    </NavigationContainer>
  );
}
