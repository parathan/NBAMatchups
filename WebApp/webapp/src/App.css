.App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

:root {
  /* --secondary-colour: #0F0F0F;
  --background-colour: #00193b;
  --navbar-colour: #00377e;
  --header-font-colour: #00BFFF;
  --navbar-font-colour: #94E1FF;
  --p-font-colour: #94E1FF; */
  --secondary-colour: #0F0F0F;
  --background-colour: #222222;
  --navbar-colour: #444444;
  --header-font-colour: #ffffff;
  --navbar-font-colour: #dddddd;
  --p-font-colour: #dddddd;
}
