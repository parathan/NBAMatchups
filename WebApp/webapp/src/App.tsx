import React from 'react';
import logo from './logo.svg';
import './App.css';

import Home from './pages/Home';
import Matchups from './pages/Matchups';
import Prediction from './pages/Prediction';

import { BrowserRouter as Router, Routes, Route } from "react-router-dom"

function App() {
  return (
    <div>
      <Router>
        <Routes>
          <Route path="/" element={<Home/>} />
          <Route path="/matchups" element={<Matchups/>} />
          <Route path="/predictions" element={<Prediction/>} />
        </Routes>
      </Router>
    </div>
  );
}

export default App;
