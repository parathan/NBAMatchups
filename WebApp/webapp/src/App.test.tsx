import React from 'react';
import { render, screen } from '@testing-library/react';
import App from './App';
import renderer from 'react-test-renderer'

test('renders learn react link', () => {
  const component = renderer.create(
    <App />,
  )
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  // render(<App />);
  // const linkElement = screen.getByText(/learn react/i);
  // expect(linkElement).toBeInTheDocument();
});
