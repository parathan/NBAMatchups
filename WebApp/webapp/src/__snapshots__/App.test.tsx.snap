// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
<div>
  <div
    className="layout"
  >
    <div
      className="navbar"
    >
      <aside
        className="ps-sidebar-root css-17c6mwq"
        data-testid="ps-sidebar-root-test-id"
        width="250px"
      >
        <div
          className="ps-sidebar-container css-dip3t8"
          data-testid="ps-sidebar-container-test-id"
        >
          <nav
            className="ps-menu-root css-vj11vy"
          >
            <ul
              className="css-ewdv3l"
            >
              <li
                className="ps-menuitem-root css-1t8x7v1"
                disabled={false}
              >
                <a
                  className="ps-menu-button"
                  data-testid="ps-menu-button-test-id"
                  href="/"
                  onClick={[Function]}
                  tabIndex={0}
                >
                  <span
                    className="ps-menu-label css-12w9als"
                  >
                    Home
                  </span>
                </a>
              </li>
              <li
                className="ps-menuitem-root css-1t8x7v1"
                disabled={false}
              >
                <a
                  className="ps-menu-button"
                  data-testid="ps-menu-button-test-id"
                  href="/matchups"
                  onClick={[Function]}
                  tabIndex={0}
                >
                  <span
                    className="ps-menu-label css-12w9als"
                  >
                    Team Matchups
                  </span>
                </a>
              </li>
              <li
                className="ps-menuitem-root css-1t8x7v1"
                disabled={false}
              >
                <a
                  className="ps-menu-button"
                  data-testid="ps-menu-button-test-id"
                  href="/predictions"
                  onClick={[Function]}
                  tabIndex={0}
                >
                  <span
                    className="ps-menu-label css-12w9als"
                  >
                    Win/Loss Prediction
                  </span>
                </a>
              </li>
              <li
                className="ps-menuitem-root css-1t8x7v1"
                disabled={false}
              >
                <a
                  className="ps-menu-button"
                  data-testid="ps-menu-button-test-id"
                  href="/dashboard"
                  onClick={[Function]}
                  tabIndex={0}
                >
                  <span
                    className="ps-menu-label css-12w9als"
                  >
                    Dashboard
                  </span>
                </a>
              </li>
            </ul>
          </nav>
        </div>
      </aside>
    </div>
    <div
      className="children"
    >
      <div
        className="homePage"
      >
        <header
          className="header"
        >
          <p>
            Welcome to NBAMatchups
          </p>
        </header>
        <div
          className="features"
        >
          <div
            className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
          >
            <div
              className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
            />
            <div
              className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
            >
              <div
                className="description"
              >
                Matchups is a feature that allows you to compare two teams based on their statistics in a given season. Just select the two teams you want to compare, and the season that you want to use. However, instead of comparing both teams on their same stats, this feature compares them based on opposing stats, and orders them by biggest differences. For example, if Team A is in the bottom percentile for defensive rebounding and Team B is in the top percentile of offensive rebounding, these are the stats that would be compared and shown near the top as the difference is the greatest. This feature would show the list of these opposing stats and order them based on these differences.
              </div>
              <button
                className="featureButton"
                onClick={[Function]}
              >
                Go to Team Matchups
              </button>
              <div
                className="description"
              >
                This prediction feature is based on a model that uses previous basketball data and the correlation these statistics have with a team's winning or losing. Using this, given two team's data, this feature predicts which team will win the matchup.
              </div>
              <button
                className="featureButton"
                onClick={[Function]}
              >
                Win/Loss Prediction
              </button>
              <div
                className="description"
              >
                The Dashboard feature visualizes the trends teams have had in the past few years for the various statistical categories that have been used in this platform. It also allows you to see the comparison between the teams statistical trends for a particular stat against the league average for this stat over the years.
              </div>
              <button
                className="featureButton"
                onClick={[Function]}
              >
                Dashboard
              </button>
            </div>
            <div
              className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
